test/library/metrics/query.go

```diff
@@ -0,0 +1,73 @@
+package metrics
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/transport"
+
+	prometheusapi "github.com/prometheus/client_golang/api"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+)
+
+// NewPrometheusClient returns Prometheus API or error
+func NewPrometheusClient(kclient *kubernetes.Clientset, rc *routeclient.Clientset) (prometheusv1.API, error) {
+	_, err := kclient.CoreV1().Services("openshift-monitoring").Get("prometheus-k8s", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	route, err := rc.RouteV1().Routes("openshift-monitoring").Get("prometheus-k8s", metav1.GetOptions{})
```

Comment:
Not something to block merge since the code works as-is., but I seem to recall @s-urbaniak mentioning that it would be preferable to query `thanos-querier`. However, my naive attempts to use the `thanos-querier` route were unsuccessful with tokens sourced from `prometheus-k8s-` and `thanos-querier-` prefixed secrets.

Reply:
Indeed, the `prometheus-k8s` is route is still maintained, but effectively deprecated. Thanos Querier is gated by the same oauth-proxy sidecar as `prometheus-k8s`, hence the same token secrets should apply, else we have a bug on our side.

Reply:
@s-urbaniak No bug. I was requesting targets via the `thanos-querier` route and it wasn't succeeding, but I realize now that it was a 404 rather than an auth failure presumably because thanos doesn't support querying targets.  Performing an `up` query instead works fine.

@sallyom Please update to use the `thanos-querier` route.

Reply:
I hit a snag, the thanos-querier route works for a query, but when searching for an alert, that route does not work, have to use the prometheus-k8s route for alerts (like I'm doing in the [kube-controller-manager-operator PR](https://github.com/openshift/cluster-kube-controller-manager-operator/pull/316)).  I added a switch and a param to the NewPrometheusClient to pass either 'query' or 'alert'.  
@s-urbaniak is that expected? thanos-querier route not for Alerts? I get a 404 for /alerts

Reply:
I found that you can pass the _entire alert_ as a query, like so:
`ALERTS{alertname="PodDisruptionBudgetAtLimit",alertstate="pending",namespace="pdbns",poddisruptionbudget="pdbname",prometheus="openshift-monitoring/k8s",service="kube-state-metrics",severity="warning"}==1`
instead of using the deprecated prometheus-k8s route.  From what I can find, you have to pass the full alert, since partial responses return an error w/ how we have thanos setup - that is stretching my knowledge of prometheus/thanos, correct me if you can.

Reply:
@sallyom do you need to pass all of the properties in that query or only those you care about? Seems quite cumbersome at first.

Reply:
@sallyom indeed, thanos querier does not support `/alerts` as it is "only" responsible for the query part. It doesn't scrape targets either as it is "just" a proxy in front of prometheus, hence `/targets` doesn't exist here either. Your `ALERTS` query is fine from my point of view, but passing just `ALERTS{alertname="PodDisruptionBudgetAtLimit"}` should work too.